### DISCUSSION TITLE
Unable to infer GraphQL type from TypeScript reflection system

### DISCUSSION THREAD

**Discussion by AlexanderFagrelius** [MichalLytek/type-graphql#1050] [Q&A]
Hello!

I am trying to add "motorid" to the accessory mutation so i can add it to the joined table on the MySQL database via Apollo server but i am getting the error  `Unable to infer GraphQL type from TypeScript reflection system. You need to provide explicit type for argument named 'motorId' of 'Accessory' of 'AccessoryResolver' class.` 

The Accessory entity and modules
![AccessoryDubble](https://user-images.githubusercontent.com/89841016/135232033-5c88a3a7-aced-4c22-b911-e1b84ba46e18.png)

The Motor entity and modules
![MotorDubble](https://user-images.githubusercontent.com/89841016/135232131-18423a48-71bb-434f-9361-7bd3d6c85703.png)

**Comment by MichalLytek** [comment]
`@Arg("motorId", () => [Motor])`

**Reply by AlexanderFagrelius** [reply]
Hello! thanks for the answer, i tried doing `@Arg("motorId", () => [Motor])` but it gave me error, 
![motoriderror](https://user-images.githubusercontent.com/89841016/135296305-943610fd-7bf9-4901-86d9-c05945b4c648.png)

i also tried doing `@Arg("motorid", () => [Motor]) access: Motor[]` but then i got the same error as before

**Reply by MichalLytek** [reply]
Dude, you've forgotten about the TS name of parameter and type 😅 

Paste your code instead of screenshots so that I can edit it, if you don't understand TypeGraphQL basics and what does the `() => [Motor]` do.

**Reply by AlexanderFagrelius** [reply]
Sorry im very bad at this, we got thrown in to a project that started with Js/React, now we are doing TS/React/ChakraUI/TypeOrm/Mysql/Type-Graphql/Apollo-server/Client

```TSX
  @Mutation(() => Accessories)
  async Accessory(
    @Arg("accessoryName") accessoryName: string,
    @Arg("motorId", () => [Motor]) motorid: Motor[]
  ): Promise<Accessories> {
    const accessory = await Accessories.create({
      accessoryName,
      motorid,
    }).save();
    return accessory;
  }

```

**Reply by MichalLytek** [reply]
And what is your error when you have `@Arg("motorId", () => [Motor]) motorid: Motor[]`?

**Reply by AlexanderFagrelius** [reply]
i get `UnhandledPromiseRejectionWarning: Error: Cannot determine GraphQL input type for argument named 'motorId' of 'Accessory' of 'AccessoryResolver' class. Is the value, that is used as its TS type or explicit type, decorated with a proper decorator or is it a proper input value?`

**Reply by MichalLytek** [reply]
So have you checked, accordingly to the error, if `Motor` class has `@InputType` decorator?

**Reply by MichalLytek** [reply]
I think you need to start with reading the docs. Then you will understand why you need input types and object types.

**Reply by AlexanderFagrelius** [reply]
The `Motor` class is a `@ObjectType`. Feels like i've read the docs a million times, but i will read it again. Thanks for trying to help atleast 😄
